epam-promo/components/inputs/RadioInput.scss

```diff
@@ -47,80 +49,143 @@
 
-            &:global(.uui-checked) {
-                background-color: var(--background-color-checked-disabled);
-                border-color: var(--border-color-checked-disabled);
-                fill: var(--fill-color-checked-disabled);
-            }
+        :global(.uui-icon) {
+            fill: var(--fill-color-checked);
         }
 
-        :global(.uui-input-label) {
-            color: var(--caption-color-disabled);
+        &:hover,
+        &:focus {
+            :global(.uui-radioinput) {
+                border-color: var(--border-color-hovered);
+            }
+
+            :global(.uui-icon) {
+                fill: var(--fill-color-hovered);
+            }
+
+            &:global(.uui-invalid) {
+                :global(.uui-radioinput) {
+                    border-color: var(--errored-color);
+                }
+
+                :global(.uui-icon) {
+                    fill: var(--errored-color);
+                }
+            }
+
+            &:global(.uui-disabled),
+            &:global(.uui-readonly) {
+                :global(.uui-radioinput) {
+                    border-color: var(--border-color-checked-disabled);
+                }
+
+                :global(.uui-icon) {
+                    fill: var(--fill-color-checked-disabled);
+                }
+            }
         }
     }
 
+    &:global(.uui-disabled),
     &:global(.uui-readonly) {
         cursor: default;
 
         :global(.uui-radioinput) {
             background-color: var(--background-color-disabled);
             border-color: var(--border-color-disabled);
+        }
 
-            &:global(.uui-checked) {
-                background-color: var(--background-color-checked-disabled);
-                border-color: var(--border-color-checked-disabled);
+        &:hover,
+        &:focus,
+        &:global(.uui-checked) {
+            :global(.uui-radioinput) {
+                background-color: var(--background-color-disabled);
+                border-color: var(--border-color-disabled);
+            }
+
+            :global(.uui-icon) {
                 fill: var(--fill-color-checked-disabled);
             }
         }
-    }
-
-    &:global(.uui-invalid) {
-        :global(.uui-radioinput) {
-            border-color: var(--errored-color);
-            fill: var(--errored-color);
 
-            &:hover, &:active {
-                border-color: var(--errored-color);
-                fill: var(--errored-color);
+        &:global(.uui-checked) {
+            :global(.uui-radioinput) {
+                border-color: var(--border-color-checked-disabled);
             }
 
-            &:global(.uui-checked) {
-                border: 1px solid var(--errored-color);
+            :global(.uui-icon) {
+                fill: var(--fill-color-checked-disabled);
             }
         }
+    }
 
+    &:global(.uui-readonly),
+    &:global(.uui-readonly):global(.uui-checked),
+    &:global(.uui-readonly):hover,
+    &:global(.uui-readonly):focus {
```

Comment:
Do we have hove and focus styles for .uui-input-label, which we need to change by readonly and disabled mods?